src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.cs

```diff
@@ -248,6 +247,144 @@ public static void FaultException_Throws_WithFaultDetail()
         Assert.True(false, "Expected FaultException<FaultDetail> exception, but no exception thrown.");
     }
 
+    [Theory]
+    [OuterLoop]
+    public static void FaultException_Throws_FaultDetail()
+    {
+        string faultMsg = "Test Fault Exception";
+        StringBuilder errorBuilder = new StringBuilder();
+
+        try
+        {
+            BasicHttpBinding binding = new BasicHttpBinding();
+            using (ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic)))
+            {
+                IWcfService serviceProxy = factory.CreateChannel();
+                serviceProxy.TestFaults(faultMsg, true);
+            }
+        }
+        catch (Exception e)
+        {
+            if (e.GetType() != typeof(FaultException<FaultDetail>))
+            {
+                string error = string.Format("Expected exception: {0}, actual: {1}\r\n{2}",
+                                             "FaultException<FaultDetail>", e.GetType(), e.ToString());
+                if (e.InnerException != null)
+                    error += String.Format("\r\nInnerException:\r\n{0}", e.InnerException.ToString());
+                errorBuilder.AppendLine(error);
+            }
+            else
+            {
+                FaultException<FaultDetail> faultException = (FaultException<FaultDetail>)(e);
+                string actualFaultMsg = ((FaultDetail)(faultException.Detail)).Message;
+                if (actualFaultMsg != faultMsg)
+                {
+                    errorBuilder.AppendLine(string.Format("Expected Fault Message: {0}, actual: {1}", faultMsg, actualFaultMsg));
+                }
+            }
+
+            Assert.True(errorBuilder.Length == 0, string.Format("Test Scenario: FaultException_Throws_WithFaultDetail FAILED with the following errors: {0}", errorBuilder));
```

Comment:
We're trying to convert all our old StringBuilder-based tests to have a better structure.  Look at TCP_ServiceCertFailedCustomValidate_Throw_Exception in this same file , specifically the way we separate setup, execute and cleanup, plus a finally block to ensure cleanup.  This suggestion is not strictly blocking the PR, but it would be nice if new tests used the new structure.